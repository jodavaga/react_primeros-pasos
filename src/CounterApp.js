import React from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({value}) => {

    // click event
    const handleAdd = (event) => {
        event.persist();
        
        console.log(event);
        const evetType = event.type;

        // Event Pooling - https://es.reactjs.org/docs/events.html#event-pooling
        setTimeout(() => {
            console.warn(evetType);
            console.log(event);
        }, 1000);
    }

    return (
        <>
            <h1>Counter App</h1>
            <h4> {value} </h4>

            {/* <button onClick={ (e) => handleAdd(e) }> +1 </button> */}
            <button onClick={ handleAdd }> +1 </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}



export default CounterApp;