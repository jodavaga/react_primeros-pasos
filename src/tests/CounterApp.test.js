import React from 'react';
import CounterApp from '../CounterApp';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';


describe('<CounterApp /> tests', () => {

    let wrapper = shallow( <CounterApp /> );

    beforeEach(() => {
        wrapper = shallow( <CounterApp /> );
    });

    test('Should render <CounterApp /> as expected', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should render <CounterApp /> with value 100', () => {

        const valor = 100;
        const wrapper = shallow( <CounterApp value={valor}/> );

        const valorTag = wrapper.find('h4').text();

        expect(valorTag).toBe( `${valor}` );
    });

    test('Should handle click event on addBtn', () => {

        const addBtn = wrapper.find('.addBtn');

        expect(addBtn.length).toEqual(1);
        // before click
        // console.log(wrapper.find('h4').text());
        addBtn.simulate('click');
        // after
        expect(wrapper.find('h4').text()).toEqual('11');
    });

    test('Should handle click event on Substaction', () => {

        // counter should be resetted (beforeEach), if not test will fail, counter is 11.
        const btn = wrapper.find('.substractBtn');

        expect(btn.length).toEqual(1);

        btn.simulate('click');
        expect(wrapper.find('h4').text()).toEqual('9');
    });

    test('Should Reset counter', () => {

        const btn = wrapper.find('.addBtn');

        expect(btn.length).toEqual(1);
        btn.simulate('click');

        // find reset button by index
        wrapper.find('button').at(1).simulate('click');

        console.log( wrapper.find('button').at(1).html() );

        expect(wrapper.find('h4').text()).toEqual('10');
    });


})