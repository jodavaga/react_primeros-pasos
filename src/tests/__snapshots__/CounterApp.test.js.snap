// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterApp /> tests Should render <CounterApp /> as expected 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h4>
    10
  </h4>
  <button
    className="addBtn"
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    className="resetBtn"
    onClick={[Function]}
  >
     Reset 
  </button>
  <button
    className="substractBtn"
    onClick={[Function]}
  >
     -1 
  </button>
</Fragment>
`;
